modules/system/files/write-text.nix

```diff
@@ -0,0 +1,78 @@
+{ lib, mkTextDerivation }:
+
+{ config, name, ... }:
+
+with lib;
+
+let
+  fileName = file: last (splitString "/" file);
+  mkDefaultIf = cond: value: mkIf cond (mkDefault value);
+
+  drv = mkTextDerivation (fileName name) config.text;
+in
+
+{
+  options = {
+    enable = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether this file should be generated.
+        This option allows specific files to be disabled.
+      '';
+    };
+
+    text = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Text of the file.
+      '';
+    };
+
+    target = mkOption {
+      type = types.str;
+      default = "/${name}";
+      description = ''
+        Name of symlink.  Defaults to the attribute name preceded by a slash (the root directory).
+      '';
+    };
```

Comment:
Note here that I've set the `default` target value to be the name of the attribute preceded by a slash. This means that when you configure a file, it looks like this:
```nix
system.file."opt/local/bin/hello".source = lib.getExe pkgs.hello
```
and a link will be created at `/opt/local/bin/hello`. I modeled this after `environment.etc`, where I think of the attr name as the path under `/etc`. I'm open to changing it if it seems unexpected, though